lightwood/encoder/categorical/onehot.py

```diff
@@ -1,140 +1,174 @@
-                    index = self._lang.word2index[word] if word in self._lang.word2index else UNCOMMON_TOKEN
-                    encoded_word[index] = 1
-
-            ret.append(encoded_word)
+        ret = torch.zeros(size=(len(column_data), self.output_size))
+
+        for idx, word in enumerate(column_data):
+            index = self.map.get(word, None)
+
+            if index is not None:
+                ret[idx, index] = 1
+
+            if self.use_unknown and index is None:
+                ret[idx, 0] = 1
 
         return torch.Tensor(ret)
 
-    def decode(self, encoded_data, return_raw=False):
+    def decode(self, encoded_data: torch.Tensor):
+        """
+        Decodes OHE mapping into the original categories. Since this approach uses an argmax, decoding flexibly works either on logits or an explicitly OHE vector.
+
+        :param: encoded_data:
+        :returns Returns the original category names for encoded data.
+        """ # noqa
         encoded_data_list = encoded_data.tolist()
         ret = []
-        probs = []
 
         for vector in encoded_data_list:
-            # Logits and onehots are not the same in definition
-            # But this explicitly operates on logits; it will take care of
-            # the one hot (so you can pass something in the softmax logit space)
-            # But will not affect something that is already OHE.
 
             all_zeros = not np.any(vector)
-            if self.handle_unknown == "return_zeros" and all_zeros:
-                ret.append(UNCOMMON_WORD)
-            else:  # self.handle_unknown == "unknown_token"
-                ohe_index = np.argmax(vector)
-                ret.append(self._lang.index2word[ohe_index])
+            if not self.use_unknown and all_zeros:
+                ret.append(_UNCOMMON_WORD)
+            else:
+                ret.append(self.rev_map[np.argmax(vector)])
 
-            if return_raw:
-                probs.append(softmax(vector).tolist())
+        return ret
 
-        if return_raw:
-            return ret, probs, self.rev_map
-        else:
-            return ret
+    def decode_probabilities(self, encoded_data: torch.Tensor):
+        """
+        Provides decoded answers, as well as a probability assignment to each data point.
+
+        :param encoded_data: the output of a mixer model
+
+        :returns Decoded values for each data point, Probability vector for each category, and the reverse map of dimension to category name
+        """ # noqa
+        encoded_data_list = encoded_data.tolist()
+        ret = []
+        probs = []
+
+        for vector in encoded_data_list:
+            if not np.any(vector):  # Vector of all 0s -> unknown category
+                ret.append(_UNCOMMON_WORD)
+            else:
+                ret.append(self.rev_map[np.argmax(vector)])
+
+            probs.append(self._norm_vec(vector))
+
+        return ret, probs, self.rev_map
+
+    @staticmethod
+    def _norm_vec(vec: List[float]):
```

Comment:
In general, I think we prefer to just inline the method in the file rather than making it static but small style nitpick, if you like it better this way keep it.